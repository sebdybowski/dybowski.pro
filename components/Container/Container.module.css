.Container {
  height: 90vh;
  width: 100vw;
  padding: 0 11rem;
}

@media screen and (max-width: 1023px) {
  .Container {
    padding: 0 2rem;
  }
}