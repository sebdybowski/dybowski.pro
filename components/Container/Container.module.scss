@use 'styles/variables.scss' as *;

$inner-width-xl: 60%;
$inner-width-lg: 70%;
$inner-width: 80%;
$inner-width-sm: 90%;
$fluid-width: 100%;
$height: 90vh;

.Container {
  width: $inner-width;
  margin: 0 auto;
  display: flex;
  flex-direction: row;

  &--fluid {
    margin: 0;
    width: $fluid-width !important;
  }

  &--section {
    height: $height;
  }
}

@media screen and (max-width: ($breakpoint-md - 1)) {
  .Container {
    width: $inner-width-sm;
    height: auto;
    min-height: 93vh;
    flex-direction: column;
  }
}

@media screen and (min-width: $breakpoint-lg) {
  .Container {
    width: $inner-width-lg;
  }
}

@media screen and (min-width: $breakpoint-xl) {
  .Container {
    width: $inner-width-xl;
  }
}