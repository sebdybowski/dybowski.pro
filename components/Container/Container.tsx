import React, { FC, ReactNode } from 'react';
import clsx from 'clsx';

import styles from './Container.module.scss';

interface ContainerProps {
  children: ReactNode,
  className?: string,
  id?: string,
}

const Container: FC<ContainerProps> = ({ children, className, id }) => (
  <section id={id} className={clsx(styles.Container, className && className)} data-testid="Container">
    {children}
  </section>
);

export default Container;
