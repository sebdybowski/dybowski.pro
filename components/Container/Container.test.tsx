import React from 'react';
import { render } from '@testing-library/react';

import '@testing-library/jest-dom/extend-expect';
import Container from './Container';

describe('<Container />', () => {
  test('it should mount', () => {
    const { asFragment } = render(<Container>Content</Container>);
    
    expect(asFragment()).toMatchSnapshot();
  });
});
