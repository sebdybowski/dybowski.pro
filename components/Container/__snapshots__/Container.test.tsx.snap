// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> it should mount 1`] = `
<DocumentFragment>
  <section
    class="Container"
    data-testid="Container"
  >
    Content
  </section>
</DocumentFragment>
`;
